prusti-viper/src/verifier.rs

```diff
@@ -74,204 +86,318 @@ impl<'v, 'tcx> Verifier<'v, 'tcx> {
-            let mut prusti_error = error_manager.translate_verification_error(&verification_error);
 
-            // annotate with counterexample, if requested
-            if config::counterexample() {
-                if config::unsafe_core_proof(){
-                    if let Some(silicon_counterexample) = &verification_error.counterexample {
-                        if let Some(def_id) = error_manager.get_def_id(&verification_error) {
-                            let counterexample = counterexample_translation_refactored::backtranslate(
-                                &self.encoder,
-                                error_manager.position_manager(),
-                                def_id,
-                                silicon_counterexample,
-                            );
-                            prusti_error = counterexample.annotate_error(prusti_error);
-                        } else {
-                            prusti_error = prusti_error.add_note(
-                                format!(
-                                    "the verifier produced a counterexample for {method}, but it could not be mapped to source code"
-                                ),
-                                None,
-                            );
-                        }
-                    }
-                } else if let Some(silicon_counterexample) = &verification_error.counterexample {
-                    if let Some(def_id) = error_manager.get_def_id(&verification_error) {
-                        let counterexample = counterexample_translation::backtranslate(
-                            &self.encoder,
-                            def_id,
-                            silicon_counterexample,
-                        );
-                        prusti_error = counterexample.annotate_error(prusti_error);
-                    } else {
-                        prusti_error = prusti_error.add_note(
+        pin_mut!(verification_messages);
+
+        while let Some((program_name, server_msg)) = verification_messages.next().await {
+            match server_msg {
+                ServerMessage::Termination(result) => {
+                    if config::show_ide_info() {
+                        PrustiError::message(
                             format!(
-                                "the verifier produced a counterexample for {method}, but it could not be mapped to source code"
+                                "ideVerificationResult{}",
+                                serde_json::to_string(&IdeVerificationResult::from_res(&result))
+                                    .unwrap()
                             ),
-                            None,
-                        );
+                            DUMMY_SP.into(),
+                        )
+                        .emit(&self.env.diagnostic);
+                    }
+                    match result.result_type {
```

Comment:
The `match` arms here are quite deeply nested. Does it make sense to extract handling for the cases into separate methods? It would also be good to be able to quickly see how much of the scope is actually needed for each (i.e. just the result kind + `prusti_errors`? `self`? `quantifier_instantiations`?).